evolver/app/routers/hardware.py

```diff
@@ -139,6 +145,19 @@ def get_calibrator_state(hardware_name: str, request: Request):
     return calibration_procedure.get_state()
 
 
+# Undo the last calibration procedure action, reverting the state to the previous state
+@router.post("/{hardware_name}/calibrator/procedure/undo")
+def undo_calibration_procedure_action(hardware_name: str, request: Request):
+    hardware_instance = get_hardware_instance(request, hardware_name)
+    calibrator = hardware_instance.calibrator
+    if not calibrator:
+        raise CalibratorNotFoundError
```

Comment:
[minor] Would be a good place for walrus!
```py
    if not (calibrator := hardware_instance.calibrator):
        raise CalibratorNotFoundError
```

Reply:
Nice, will fix.